server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -53,6 +59,10 @@ public static DeprecationLogger getLogger(String name) {
         return new DeprecationLogger(name);
     }
 
+    public static void setInitialEnvironmentSettings(Settings settings) {
+        initialEnvironment = settings;
```

Comment:
Super minor, but we call these "nodeSettings" in a few places to distinguish them from generic `Settings` objects, might help clarify here (as well as adding javadoc)